### PR TITLE
Modify the expression about the error check

### PR DESCRIPTION
**Pull request by yuexiao-wang**
The  "err != nil " is redundant when matching error condition. So, remove the  check about "err != nil "  

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

### CODE REVIEW

File: pkg/auth/server/session/session.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func NewStore(secure bool, maxAgeSeconds int, secrets ...string) Store {
 
 func (s store) Get(req *http.Request, name string) (Session, error) {
 	session, err := s.store.Get(req, name)
-	if err != nil && err.Error() == securecookie.ErrMacInvalid.Error() {
+	if err.Error() == securecookie.ErrMacInvalid.Error() {
```
Comment (deads2k) [inline pkg/auth/server/session/session.go]: `err.Error()` can (but doesn't always) panic depending on how the particular `error` was written to handle `nil` method receivers.

File: pkg/router/f5/f5.go [hunk 2]
```diff
@@ -470,7 +470,7 @@ func (f5 *f5LTM) ensurePolicyExists(policyName string) error {
 		f5.host, policyName)
 
 	err := f5.get(policyUrl, nil)
-	if err != nil && err.(F5Error).httpStatusCode != 404 {
+	if err.(F5Error).httpStatusCode != 404 {
```
Comment (deads2k) [inline pkg/router/f5/f5.go]: You aren't supposed to perform type assertions on `nil` values (https://golang.org/ref/spec#Type_assertions).  `error` maybe special (I don't know for sure), but this change still reads incorrectly to me.

File: pkg/generate/app/sourcelookup.go [hunk 3]
```diff
@@ -260,7 +260,7 @@ func (r *SourceRepository) RemoteURL() (*url.URL, bool, error) {
 	case "file":
 		gitRepo := git.NewRepository()
 		remote, ok, err := gitRepo.GetOriginURL(r.url.Path)
-		if err != nil && err != git.ErrGitNotAvailable {
+		if err != git.ErrGitNotAvailable {
```
Comment (deads2k) [inline pkg/generate/app/sourcelookup.go]: This changes the behavior of the method.  With this change if you have a `nil` error and `ok == true`, you'll return here with an incorrect (`false`) return value.

File: pkg/cmd/util/env.go [hunk 4]
```diff
@@ -152,7 +152,7 @@ func readEnv(r io.Reader) ([]kapi.EnvVar, error) {
 			})
 		}
 	}
-	if err := scanner.Err(); err != nil && err != io.EOF {
+	if err := scanner.Err(); err != io.EOF {
```
Comment (liggitt) [inline pkg/cmd/util/env.go]: this is wrong... it means if err is nil, we won't return the env

**Comment by liggitt** [comment]
we want to keep the nil checks on these error conditions

**Pull request closed without merging**